docs/table_configuration.rst

```diff
@@ -0,0 +1,220 @@
+
+Profile conditionals
+--------------------
+
+Configuration values can be set based on which profiles are enabled. For example:
+
+.. code-block:: javascript
+
+  table {
+    //...
+    replication___prod = 3
+    replication___testing = 1
+  }
+
+In the example above, the replication for the table will be 3 if the ``prod`` profile is enabled, 1 if the ``testing``
+profile is enabled, and unset otherwise.
+
+There is no conflict resolution between conflicting conditional values, so a configuration containing simultaneous
+different values for profile conditionals is invalid.
+
+.. code-block:: javascript
+
+  table {
+    // Invalid value for replication, it simultaneously has a value of 3 and 4
+    // if the prod and prod-us-west-4 profiles are enabled at the same time
+    replication___prod = 3
+    replication___prod-us-west-4 = 4
+  }
+
+Different settings for realtime and offline
+-------------------------------------------
+
+Different settings for realtime and offline tables can be set by adding a ``offline`` or ``realtime`` suffix to the
+property value. If not specified, the property value applies to both the offline and realtime counterparts of a table.
+
+.. code-block:: javascript
+
+  table {
+    // ...
+    "retention.offline": "90 days"
+    "retention.realtime": "5 days"
+  }
+
+This can be combined with profile conditionals as follows:
+
+.. code-block:: javascript
+
+  table {
+    // ...
+    "retention.offline___testing": "14 days"
+    "retention.offline___prod": "90 days"
+    "retention.realtime": "5 days"
+  }
+
+Operations
+==========
+
+Deploying configurations
+------------------------
+
+Configurations can be deployed using the Pinot admin tools, as follows:
+
+.. code-block:: bash
+
+  pinot-admin.sh ApplyTableConfig -tableConfigFile my-table.conf -profile prod,prod-us-west-4 \
+      -controllerUrl http://pinot-controller-proxy.us-west-4.cloudprovider.com:23456/
+
+For batch automatic deployments, the table configuration file parameter can be a directory, which will recursively process all the table
+definition files in that directory.
+
+We recommend managing table configurations using a source control system and automating table deployments by running the
+``ApplyTableConfig`` command on source control changes or periodically using a scheduler.
+
+Backing up table configurations
+-------------------------------
+
+Table configurations can be backed up using the ``BackupTableConfigs`` command:
+
+.. code-block:: bash
+
+  pinot/bin/pinot-admin.sh BackupTableConfigs -tableName myTable \
+      -controllerUrl http://pinot-controller-proxy.us-west-4.cloudprovider.com:23456/
+
+Migrating from the legacy configuration format
+----------------------------------------------
+
+For single cluster Pinot deployments, migration is done by backing up all of the table configurations as shown above.
+
+For multiple cluster Pinot deployments, an additional merge step needs to be done to minimize the number of table files.
```

Comment:
It will be useful to expand this a bit. I suppose the merge step is really not needed, but is useful (as you have pointed out, to minimize the number of files).
I would add a step between 1 and 2 that says something like "Create a folder structure depending on your deployments. For example, you can have folders configBackup/prod/prod-us-west-4, configBackup/prod/prod-us-east-3"
Expanding step 2 a bit is also useful. As in "Now change directory to each of the folders for specific deployment and enter the BackupTableConfigs command"

Some clarifications:
Is there a requirement that the 'profiles' directory be under configBackup/ some place? Can you document it either way?